hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/QuotaRepairTask.java

```diff
@@ -0,0 +1,326 @@
+            bucketKey, entry.getValue());
+      }
+      metadataManager.getStore().commitBatchOperation(batchOperation);
+    }
+    LOG.info("Completed quota repair for all keys, files and directories");
+  }
+  
+  private <VALUE> void recalculateUsages(
+      Table<String, VALUE> table, Map<String, CountPair> prefixUsageMap,
+      String strType, boolean haveValue) throws UncheckedIOException,
+      UncheckedExecutionException {
+    LOG.info("Starting recalculate {}", strType);
+
+    List<Table.KeyValue<String, VALUE>> kvList = new ArrayList<>(BATCH_SIZE);
+    BlockingQueue<List<Table.KeyValue<String, VALUE>>> q
+        = new ArrayBlockingQueue<>(TASK_THREAD_CNT);
+    List<Future<?>> tasks = new ArrayList<>();
+    AtomicBoolean isRunning = new AtomicBoolean(true);
+    for (int i = 0; i < TASK_THREAD_CNT; ++i) {
+      tasks.add(executor.submit(() -> captureCount(
+          prefixUsageMap, q, isRunning, haveValue)));
+    }
+    int count = 0;
+    long startTime = System.currentTimeMillis();
+    try (TableIterator<String, ? extends Table.KeyValue<String, VALUE>>
+             keyIter = table.iterator()) {
+      while (keyIter.hasNext()) {
+        count++;
+        kvList.add(keyIter.next());
+        if (kvList.size() == BATCH_SIZE) {
+          q.put(kvList);
+          kvList = new ArrayList<>(BATCH_SIZE);
+        }
+      }
+      q.put(kvList);
+      isRunning.set(false);
+      for (Future<?> f : tasks) {
+        f.get();
+      }
+      LOG.info("Recalculate {} completed, count {} time {}ms", strType,
+          count, (System.currentTimeMillis() - startTime));
+    } catch (IOException ex) {
+      throw new UncheckedIOException(ex);
+    } catch (InterruptedException ex) {
+      Thread.currentThread().interrupt();
+    } catch (ExecutionException ex) {
+      throw new UncheckedExecutionException(ex);
+    }
+  }
+  
+  private <VALUE> void captureCount(
+      Map<String, CountPair> prefixUsageMap,
+      BlockingQueue<List<Table.KeyValue<String, VALUE>>> q,
+      AtomicBoolean isRunning, boolean haveValue) throws UncheckedIOException {
+    try {
+      while (isRunning.get() || !q.isEmpty()) {
+        List<Table.KeyValue<String, VALUE>> kvList
+            = q.poll(100, TimeUnit.MILLISECONDS);
+        if (null != kvList) {
+          for (Table.KeyValue<String, VALUE> kv : kvList) {
+            extractCount(kv, prefixUsageMap, haveValue);
+          }
+        }
+      }
+    } catch (InterruptedException ex) {
+      Thread.currentThread().interrupt();
+    }
+  }
+  
+  private <VALUE> void extractCount(
+      Table.KeyValue<String, VALUE> kv,
+      Map<String, CountPair> prefixUsageMap,
+      boolean haveValue) {
+    try {
+      String prefix = getVolumeBucketPrefix(kv.getKey());
+      CountPair usage = prefixUsageMap.get(prefix);
+      if (null == usage) {
+        return;
+      }
+      usage.incrNamespace(1L);
+      // avoid decode of value
+      if (haveValue) {
```

Comment:
@sumitagrawl, This may not work since `OmKeyInfo`/`OmDirectoryInfo` supported CodecBuffer -- they were always decoded.  The correct way is to support key only iterator.  Cc @swamirishi , @adoroszlai